drivers/builtin/include/mbedtls/cipher.h

```diff
@@ -1164,6 +1167,9 @@ int mbedtls_cipher_auth_decrypt_ext(mbedtls_cipher_context_t *ctx,
                                     unsigned char *output, size_t output_len,
                                     size_t *olen, size_t tag_len);
 #endif /* MBEDTLS_CIPHER_MODE_AEAD || MBEDTLS_NIST_KW_C */
+
+#endif /* MBEDTLS_DECLARE_PRIVATE_IDENTIFIERS */
```

Comment:
It would be convenient to additionally annotate all function declarations between `#if defined(MBEDTLS_DECLARE_PRIVATE_IDENTIFIERS)` and the corresponding `#endif` with `MBEDTLS_DECLARE_PRIVATE_FUNCTION` which can be defined to expand to something like `__attribute__((__deprecated__))`. This would allow detecting the uses of private functions through compiler warnings, and gradually eliminating them by eliminating warnings.

We can add these annotations mechanically, once we've manually added the preprocessor guards.